runtime/compiler/env/J2IThunk.hpp

```diff
@@ -72,7 +72,9 @@ class TR_J2IThunkTable
          case 'J': return TC_LONG;
          case 'F': return TC_FLOAT;
          case 'D': return TC_DOUBLE;
-         case 'L': return TC_REFERENCE;
+         case 'L': 
+         case 'Q': //TODO: further investigation is needed for 'Q'
```

Comment:
As above, we should track the investigation that's needed in a new work item instead of marking this with a TODO.  Sorry for suggesting previously that you should use a TODO!